.github/workflows/ci.yaml

```diff
@@ -15,9 +15,11 @@ jobs:
       - uses: extractions/setup-just@v2
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - uses: actions/setup-python@v5
-      - uses: hynek/setup-cached-uv@v1
       - uses: actions/checkout@v4
+      - uses: astral-sh/setup-uv@v6
+        with:
+          activate-environment: true
+          python-version: 3.13
```

Comment:
Should we still use `UV_SYSTEM_PYTHON`? I just saw that you removed it from ruff-vscode CI pipeline